config/config.go

```diff
@@ -37,16 +39,103 @@ var (
 	ErrRemoteConfigEmptyName = errors.New("remote config: empty name")
 )
 
+// Options allows configuration of the config package
+type Options struct {
+	// fs provides a configurable entry point to the billy.Filesystem in active use for reading global and
+	// system level configuration; set using the WithFS option in a test to use a fake filesystem.
+	fs billy.Filesystem
+}
+
+func (o Options) readConfigFromFS(path string) (*Config, error) {
+	f, err := o.fs.Open(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, nil
+		}
+
+		return nil, err
+	}
+
+	defer f.Close()
+	return ReadConfig(f)
+}
+
+func NewOptions(ofns ...WithOptions) Options {
+	o := Options{
+		fs: DefaultFS(),
+	}
+
+	for _, fn := range ofns {
+		fn(&o)
+	}
+
+	return o
+}
+
+type WithOptions = func(*Options)
+
+func WithFS(fs billy.Filesystem) func(*Options) {
```

Comment:
You could just refer to the type you defined:

```suggestion
func WithFS(fs billy.Filesystem) Option {
```